python/tests/unit/processes/local_runtime/test_local_step.py

```diff
@@ -0,0 +1,558 @@
+    mock_info = KernelProcessStepInfo(
+        inner_step_type=AsyncMock(spec=type),
+        state=mock_state,
+        output_edges={"test_event": [edge]},
+    )
+    step = LocalStep(
+        kernel=AsyncMock(spec=Kernel),  # type: ignore
+        step_info=mock_info,
+        event_namespace="ns",
+        step_state=mock_state,
+        factories={},
+    )
+
+    edges = step.get_edge_for_event("test_event")
+    assert len(edges) == 1
+    assert edges[0] == edge
+
+    # For a non-existing event, expect empty list
+    assert step.get_edge_for_event("not_found") == []
+
+
+async def test_to_kernel_process_step_info_initializes_if_needed():
+    """Test to_kernel_process_step_info calls initialize_step if not yet done."""
+    mock_state = KernelProcessStepState(name="my_step", id="123", state=None)
+    mock_info = KernelProcessStepInfo(
+        inner_step_type=AsyncMock(spec=type),
+        state=mock_state,
+        output_edges={},
+    )
+    step = LocalStep(
+        kernel=AsyncMock(spec=Kernel),
+        step_info=mock_info,
+        event_namespace="ns",
+        step_state=mock_state,
+        factories={},
+        initialize_task=False,
+    )
+
+    with patch.object(LocalStep, "initialize_step") as mock_initialize_step:
+        result = await step.to_kernel_process_step_info()
+
+        mock_initialize_step.assert_awaited_once()
+        assert result == mock_info
+
+
+def test_scoped_event_updates_namespace():
+    """Test scoped_event sets the local_event's namespace to name_id."""
+    mock_state = KernelProcessStepState(name="my_step", id="123", state=None)
+    mock_info = KernelProcessStepInfo(
+        inner_step_type=AsyncMock(spec=type),
+        state=mock_state,
+        output_edges={},
+    )
+    step = LocalStep(
+        kernel=AsyncMock(spec=Kernel),
+        step_info=mock_info,
+        event_namespace="ns",
+        step_state=mock_state,
+        factories={},
+    )
+
+    some_event = LocalEvent(namespace=None, inner_event=KernelProcessEvent(id="evt"))
+    result = step.scoped_event(some_event)
+
+    assert result.namespace == "my_step_123"
+
+
+def test_scoped_event_from_kernel_process_creates_scoped_event():
+    """Test scoped_event_from_kernel_process creates a local event from the kernel process event
+    with the step's scope."""
+    mock_state = KernelProcessStepState(name="my_step", id="321", state=None)
+    mock_info = KernelProcessStepInfo(
+        inner_step_type=AsyncMock(spec=type),
+        state=mock_state,
+        output_edges={},
+    )
+    step = LocalStep(
+        kernel=AsyncMock(spec=Kernel),  # type: ignore
+        step_info=mock_info,
+        event_namespace="ns",
+        step_state=mock_state,
+        factories={},
+    )
+
+    kpe = KernelProcessEvent(id="test_id", data="some_data")
+    local_event = step.scoped_event_from_kernel_process(kpe)
+
+    assert local_event.namespace == "my_step_321"
+    assert local_event.inner_event == kpe
```

Comment:
We could also assert that `inner_event` is the same object (not a copy) to make sure there's no accidental duplication.